### ISSUE TITLE
Different behavior when too large value is assigned to int array

### ISSUE THREAD

**Issue by unnonouno** [pfnet/chainer#815]
```
>>> x=cupy.array([1], dtype=cupy.int32)
>>> x+=float('inf')
>>> x
array([2147483647], dtype=int32)
>>> y=numpy.array([1], dtype=numpy.int32)
>>> y+=float('inf')
>>> y
array([-2147483648], dtype=int32)
```

Cupy array has max value of `int32`, though numpy array has min value of `int32`.

**Comment by unnonouno** [comment]
We need to make a list of difference between CuPy and NumPy. related to #610

**Comment by unnonouno** [comment]
fixed via #1723
